video/out/vo_gpu_next.c

```diff
@@ -842,7 +842,7 @@ static void apply_target_options(struct priv *p, struct pl_frame *target)
     int dither_depth = opts->dither_depth;
     if (dither_depth == 0) {
         struct ra_swapchain *sw = p->ra_ctx->swapchain;
-        if (sw->fns->color_depth) {
+        if (sw->fns->color_depth && sw->fns->color_depth(sw) != 0) {
```

Comment:
Shouldn't need to check that. Anybody implementing this but returning 0 or some other bad value is just wrong.

Reply:
0 is also perfectly valid value default value for auto dither.

Reply:
if we only check for `sw->fns->color_depth` current behaviour will be changed. see also https://github.com/mpv-player/mpv/pull/13643 where we removed the `color_depth(struct ra_swapchain *sw)` stub in `video/out/vulkan/context.c`.

now that i added the 'stub' back for vulkan, at least it always returns 0 on all other contexts, the `if (sw->fns->color_depth)` would always be true and the `else if (!pl_color_transfer_is_hdr(target->color.transfer))` condition could never be hit in any case. making it basically dead code.

since `dither_depth == 0` is checked before, `dither_depth` is already 0 in that case. i decided to only change it if the actual function call would change it, and the `else if` can still be hit.

would you prefer me to change the if/else if to something like?

```
    struct ra_swapchain *sw = p->ra_ctx->swapchain;
    if (dither_depth == 0 && sw->fns->color_depth)
        dither_depth = sw->fns->color_depth(sw);
    if (dither_depth == 0 && !pl_color_transfer_is_hdr(target->color.transfer))
        dither_depth = 8;
```